src/Spago/Config.hs

```diff
@@ -398,6 +410,26 @@ withConfigAST transform = do
         else logDebug "Transformed config is the same as the read one, not overwriting it"
       pure exprHasChanged
 
+-- | Takes a function that manipulates the Dhall AST of the Config, and tries to run it
+--   on the current config. If it succeeds, it writes back to file the result returned.
+withRawConfigAST
+  :: (HasLogFunc env, HasConfigPath env)
+  => (ResolvedExpr -> Expr -> RIO env Expr) -> RIO env Bool
+withRawConfigAST transform = do
+  ConfigPath path <- view (the @ConfigPath)
+  rawConfig <- liftIO $ Dhall.readRawExpr path
+  normalizedExpr <- liftIO $ Dhall.inputExpr $ "./" <> path
+  case rawConfig of
+    Nothing -> die [ display $ Messages.cannotFindConfig path ]
+    Just (header, expr) -> do
+      newExpr <- transform normalizedExpr $ Dhall.Core.denote expr
```

Comment:
I think newtyping a tuple would be better because it clarifies that the normalized and raw expression are the same expression in two different versions. A simple tuple does not do that.

Reply:
Yeah this is great!